src/transforms/groupby.js

```diff
@@ -163,12 +177,14 @@ function transformOne(trace, state) {
 
         var newTrace = newData[i] = Lib.extendDeepNoArrays({}, trace);
 
+        cloneTransforms(newTrace);
+
         arrayAttrs.forEach(initializeArray.bind(null, newTrace));
 
-        for(var j = 0; j < len; j++) {
+        for(j = 0; j < len; j++) {
             if(groups[j] !== groupName) continue;
 
-            arrayAttrs.forEach(pasteArray.bind(0, newTrace, trace, j));
+            arrayAttrs.forEach(pasteArray.bind(null, newTrace, trace, j));
```

Comment:
This isn't new here - except that the change you made reminded me how `bind` always confuses me - but it occurs to me that there's no reason to use it at all if we unwind the `forEach` to a `for`. Also `pasteArray` is a bit inefficient in the way it calls `Lib.nestedProperty(newTrace, a)` twice.